### PR TITLE
Add support for the Apalis iMX6 target

### PR DESCRIPTION
**Pull request by EmbeddedAndroid**
This series adds support to the Linux microPlatform for the Apalis iMX6 SoM from Toradex for use with the Ixora carrier board.

### CODE REVIEW

**Comment by EmbeddedAndroid** [comment]
If we can land these changes: https://github.com/foundriesio/lmp-kernel-cache/pull/1 then I'll bump the kmeta hash when this looks ready. Need to give this branch a test locally.

**Comment by EmbeddedAndroid** [comment]
Tested locally on hardware, good to go.

File: classes/lmp-machine-custom.bbclass [hunk 1]
```diff
@@ -102,6 +102,17 @@ UBOOT_DTB_LOADADDRESS_cl-som-imx7 = "0x82000000"
 UBOOT_RD_LOADADDRESS_cl-som-imx7 = "0x82100000"
 WKS_FILE_sota_cl-som-imx7 = "sdimage-imx7-spl-sota.wks"
 
+# Toradex Apalis iMX6 (support both NAND and eMMC targets with one single image)
+OSTREE_KERNEL_ARGS_append_apalis-imx6 = " console=tty1 console=ttymxc0,115200"
+EXTRA_IMAGEDEPENDS_append_apalis-imx6 = " u-boot-script-toradex"
+KERNEL_DEVICETREE_append_apalis_imx6 = "imx6q-apalis-ixora-v1.1.dtb"
```
Comment (ricardosalveti) [inline classes/lmp-machine-custom.bbclass]: This shouldn't be needed as the machine configuration file already has such dtb in KERNEL_DEVICETREE.

File: recipes-bsp/u-boot/u-boot-toradex/0001-apalis_imx6-run-distro-bootcmd-by-default.patch [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
++#define FDT_FILE "imx6q-apalis-ixora-v1.1.dtb"
+ #define FDT_FILE_V1_0 "imx6q-apalis_v1_0-eval.dtb"
+ #else
+ #define FDT_FILE "imx6q-apalis_v1_0-eval.dtb"
+@@ -265,7 +266,7 @@
+ 	"defargs=vmalloc=400M user_debug=30\0" \
+ 	"dfu_alt_info=" DFU_ALT_EMMC_INFO "\0" \
+ 	EMMC_BOOTCMD \
+-	"fdt_file=" FDT_FILE "\0" \
++	"fdtfile=" FDT_FILE "\0" \
```
Comment (ricardosalveti) [inline recipes-bsp/u-boot/u-boot-toradex/0001-apalis_imx6-run-distro-bootcmd-by-default.patch]: Is this change needed? You are already setting fdtfile via CONFIG_BOOTCOMMAND.

File: classes/lmp-machine-custom.bbclass (hunk 1 above)
Comment (EmbeddedAndroid) [reply classes/lmp-machine-custom.bbclass]: Ok I've dropped this line and the machine conf in favor of the upstream one.

File: recipes-bsp/u-boot/u-boot-toradex/0001-apalis_imx6-run-distro-bootcmd-by-default.patch (hunk 2 above)
Comment (EmbeddedAndroid) [reply recipes-bsp/u-boot/u-boot-toradex/0001-apalis_imx6-run-distro-bootcmd-by-default.patch]: Was following the patch you did for the imx7, but no it's not needed. I've dropped this bit in the latest series.

File: recipes-bsp/u-boot/u-boot-toradex/0001-apalis_imx6-run-distro-bootcmd-by-default.patch [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+diff --git a/include/configs/apalis_imx6.h b/include/configs/apalis_imx6.h
+index 71dfeb56f5..54219bc103 100644
+--- a/include/configs/apalis_imx6.h
++++ b/include/configs/apalis_imx6.h
+@@ -180,6 +180,7 @@
+         func(DHCP, dhcp, na)
+ #include <config_distro_bootcmd.h>
+ #include <config_distro_defaults.h>
++#define CONFIG_BOOTCOMMAND "setenv fdtfile imx6q-apalis-ixora-v1.1.dtb && run distro_bootcmd;
```
Comment (ricardosalveti) [inline recipes-bsp/u-boot/u-boot-toradex/0001-apalis_imx6-run-distro-bootcmd-by-default.patch]: Missing " at the end of the line.

```
git/inc
lude/configs/apalis_imx6.h:183:28: warning: missing terminating " character
|  #define CONFIG_BOOTCOMMAND "setenv fdtfile imx6q-apalis-ixora-v1.1.dtb && run distro_bootcmd;
```

Comment (EmbeddedAndroid) [reply recipes-bsp/u-boot/u-boot-toradex/0001-apalis_imx6-run-distro-bootcmd-by-default.patch]: Good catch, let me fix that.

Comment (EmbeddedAndroid) [reply recipes-bsp/u-boot/u-boot-toradex/0001-apalis_imx6-run-distro-bootcmd-by-default.patch]: updated
